### PR TITLE
cloud Data Service: add API key role, linked clusters, and update the endpoint response

### PR DESCRIPTION
**Pull request by Oreoxmt**
### What is changed, added or deleted? (Required)

- Update the endpoint response: remove query and update message
- Update the max value of chat2data timeout from 30000 to 60000
- Add API key role: ReadOnly and ReadAndWrite
- Add data source: link a cluster as the data source or remove a cluster from the data source

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tidb-cloud/data-service-api-key.md [hunk 1]
```diff
@@ -32,11 +32,20 @@ To create an API key for a Data App, perform the following steps:
...
 3. In the **API Key** area, click **Create API Key**.
-4. In the **Create API Key** dialog box, enter a description for your API key, and then click **Next**. The private key and public key are displayed.
+<!--TODO: add API key role (ReadOnly and ReadAndWrite)-->
+4. In the **Create API Key** dialog box, enter a description and select a role for your API key.
+
+    - The description field is optional and can help you identify an API key.
+    - The role is used to control the permission of the API key to access clusters linked to the Data App. You can select `ReadOnly` or `ReadAndWrite` as the role:<!--TODO: add the description of ReadOnly and ReadAndWrite-->
+
+        - `ReadOnly`: only allows the API key to read data, such as a `SELECT` statement.
+        - `ReadAndWrite`: allows the API key to read and write data. You can use this API key to execute all SQL queries, such as DML and DDL statements.
```
Comment (qiancai) [inline tidb-cloud/data-service-api-key.md:42]: Query 是查询。DML and DDL statement 是否算 SQL queries 呢？

File: tidb-cloud/data-service-get-started.md [hunk 2]
```diff
@@ -62,13 +70,15 @@ On the right pane, click the **Properties** tab and set properties for the endpo
 
 You can customize SQL statements for the endpoint in the SQL editor, which is the middle pane on the **Data Service** page.
 
-1. Select a cluster.
+<!--TODO: add a drop-down list: Add Data Source-->
+
+1. Select clusters as the data source.
```
Comment (qiancai) [inline tidb-cloud/data-service-get-started.md:75]:
```suggestion
1. Select a cluster as the data source.
```

Comment (qiancai) [reply tidb-cloud/data-service-get-started.md:75]: 一次只能选一个 cluster
image

File: tidb-cloud/data-service-get-started.md [hunk 3]
```diff
@@ -62,13 +70,15 @@ On the right pane, click the **Properties** tab and set properties for the endpo
...
 
     > **Note:**
     >
-    > Only clusters in the **Available** state are displayed in the drop-down list.
+    > Only clusters that are linked to the Data App and in the **Available** state are displayed in the drop-down list. To manage the linked clusters, see [Mange the data source](/tidb-cloud/data-service-manage-data-app.md#manage-the-data-source).
 
-    On the upper part of the SQL editor, select a cluster on which the SQL statements are executed from the drop-down list. Then, you can view all databases of this cluster in the **Schema** tab on the right pane.
+    On the upper part of the SQL editor, select one or multiple clusters from the drop-down list. Then, you can view all databases of the selected clusters in the **Schema** tab on the right pane.
```
Comment (qiancai) [inline tidb-cloud/data-service-get-started.md:81]: select one or multiple clusters，问题同上

File: tidb-cloud/data-service-manage-data-app.md [hunk 4]
```diff
@@ -33,11 +33,27 @@ In the left pane of the [**Data Service**](https://tidbcloud.com/console/dataser
     - **Public Key**: the public key of the API key.
     - **Private Key**: the private key of the API key.
     - **Description**: the description of the API key.
-    - **Action**: you can edit the description of the key or delete the key.
+    <!--TODO: add two new fields **Created Time** and **Role**-->
```
Comment (qiancai) [inline tidb-cloud/data-service-manage-data-app.md:36]: View a Data App 的操作很简单，页面字段也很理解，所以是否可以考虑直接删除 L24~L52，不影响用户使用，也方便我们维护

File: tidb-cloud/data-service-manage-data-app.md [hunk 5]
```diff
@@ -47,6 +63,34 @@ To rename a Data App, navigate to the [**Data Service**](https://tidbcloud.com/c
 
 - In the left pane, click the name of your target Data App to view its details. In the **Basic Settings** area, modify the **Name** field and click **Save**.
 
+### Manage the data source
```
Comment (qiancai) [inline tidb-cloud/data-service-manage-data-app.md:66]: 是否需要改成 Manage linked clusters

File: tidb-cloud/data-service-manage-data-app.md [hunk 6]
```diff
@@ -47,6 +63,34 @@ To rename a Data App, navigate to the [**Data Service**](https://tidbcloud.com/c
...
+
+<!--TODO: **Add Cluster**-->
+
+You can add or remove a cluster as a data source for a Data App. If a cluster in the data source is deleted, it is automatically removed from the Data App, but the endpoints in the Data App can still access other clusters in the data source.
```
Comment (qiancai) [inline tidb-cloud/data-service-manage-data-app.md:70]:
```suggestion
You can add or remove a cluster as a data source for a Data App. After a cluster is removed from the Data App, the endpoints in the Data App can still access other linked clusters.
```

File: tidb-cloud/data-service-manage-data-app.md [hunk 7]
```diff
@@ -47,6 +63,34 @@ To rename a Data App, navigate to the [**Data Service**](https://tidbcloud.com/c
...
+
+<!--TODO: add a Data Source-->
+
+To add a new cluster as the data source of a Data App, perform the following steps:
```
Comment (qiancai) [inline tidb-cloud/data-service-manage-data-app.md:74]:
```suggestion
To add a cluster as the data source of a Data App, perform the following steps:
```

File: tidb-cloud/data-service-manage-data-app.md [hunk 8]
```diff
@@ -47,6 +63,34 @@ To rename a Data App, navigate to the [**Data Service**](https://tidbcloud.com/c
...
+
+1. Navigate to the [**Data Service**](https://tidbcloud.com/console/dataservice) page of your project.
+2. In the left pane, locate your target Data App and click the name of your target Data App to view its details.
+3. In the **Data Source** area, click **Add Cluster** and the **Link Cluster** dialog box is displayed.
```
Comment (qiancai) [inline tidb-cloud/data-service-manage-data-app.md:78]:
```suggestion
3. In the **Link Cluster** area, click **Add Cluster**.
```

Comment (qiancai) [reply tidb-cloud/data-service-manage-data-app.md:78]: dialog box 的名字总是变，不写也挺容易理解的

File: tidb-cloud/data-service-manage-data-app.md [hunk 9]
```diff
@@ -47,6 +63,34 @@ To rename a Data App, navigate to the [**Data Service**](https://tidbcloud.com/c
...
+4. Select a cluster from the list and click **Confirm**.
```
Comment (qiancai) [inline tidb-cloud/data-service-manage-data-app.md:79]:
```suggestion
4. In the displayed dialog box, select a cluster from the list and click **Confirm**.
```

File: tidb-cloud/data-service-manage-data-app.md [hunk 10]
```diff
@@ -47,6 +63,34 @@ To rename a Data App, navigate to the [**Data Service**](https://tidbcloud.com/c
...
+
+    The newly linked cluster is added to the data source list.
```
Comment (qiancai) [inline tidb-cloud/data-service-manage-data-app.md:81]:
```suggestion
    > **Note:**
    >
    > Only clusters in the **Available** state are displayed in the drop-down list.
```

Comment (qiancai) [reply tidb-cloud/data-service-manage-data-app.md:81]: 显而易见的内容一般可以不写

File: tidb-cloud/data-service-manage-data-app.md [hunk 11]
```diff
@@ -47,6 +63,34 @@ To rename a Data App, navigate to the [**Data Service**](https://tidbcloud.com/c
...
+
+<!--TODO: remove a Data Source-->
+
+To remove a cluster from the data source of a Data App, perform the following steps:
+
+1. Navigate to the [**Data Service**](https://tidbcloud.com/console/dataservice) page of your project.
+2. In the left pane, locate your target Data App and click the name of your target Data App to view its details.
+3. In the **Data Source** area, locate the target cluster you want to remove from the Data App, and click **Delete** in the **Action** column.
+4. The **Remove Data Source** dialog box is displayed. Click **I understand, remove** to confirm the removal.
```
Comment (qiancai) [inline tidb-cloud/data-service-manage-data-app.md:90]:
```suggestion
4. The **Remove Data Source** dialog box is displayed. Click **I understand, remove** to confirm the removal.
```

File: tidb-cloud/data-service-api-key.md [hunk 12]
```diff
@@ -32,11 +32,19 @@ To create an API key for a Data App, perform the following steps:
 1. Navigate to the [**Data Service**](https://tidbcloud.com/console/dataservice) page of your project.
 2. In the left pane, click the name of your target Data App to view its details.
 3. In the **API Key** area, click **Create API Key**.
-4. In the **Create API Key** dialog box, enter a description for your API key, and then click **Next**. The private key and public key are displayed.
+<!--TODO: add API key role (ReadOnly and ReadAndWrite)-->
+4. In the **Create API Key** dialog box, enter a description and select a role for your API key.
+
+    The role is used to control whether the API key can read or write data to the clusters linked to the Data App. You can select the `ReadOnly` or `ReadAndWrite` role:
+
+    - `ReadOnly`: only allows the API key to read data, such as a `SELECT` statement.
```
Comment (heibaijian) [inline tidb-cloud/data-service-api-key.md:40]: 除了 select 还有 show、use、desc、explain

Comment (Oreoxmt) [inline tidb-cloud/data-service-api-key.md:40]:
```suggestion
    - `ReadOnly`: only allows the API key to read data, such as `SELECT`, `SHOW`, `USE`, `DESC`, and `EXPLAIN` statements.
```

Comment (Oreoxmt) [reply tidb-cloud/data-service-api-key.md:40]: @heibaijian PTAL

File: tidb-cloud/data-service-get-started.md [hunk 13]
```diff
@@ -62,11 +65,13 @@ On the right pane, click the **Properties** tab and set properties for the endpo
...
+<!--TODO: update the drop-down list: Select a cluster-->
+
 1. Select a cluster.
 
     > **Note:**
     >
-    > Only clusters in the **Available** state are displayed in the drop-down list.
+    > Only clusters that are linked to the Data App are displayed in the drop-down list. To manage the linked clusters, see [Manage linked clusters](/tidb-cloud/data-service-manage-data-app.md#manage-linked-clusters).
 
     On the upper part of the SQL editor, select a cluster on which the SQL statements are executed from the drop-down list. Then, you can view all databases of this cluster in the **Schema** tab on the right pane.
```
Comment (qiancai) [inline tidb-cloud/data-service-get-started.md:76]:
```suggestion
    On the upper part of the SQL editor, select a cluster on which you want to execute SQL statements from the drop-down list. Then, you can view all databases of this cluster in the **Schema** tab on the right pane.
```

File: tidb-cloud/data-service-manage-data-app.md [hunk 14]
```diff
@@ -17,35 +17,43 @@ To create a Data App for your project, perform the following steps:
...
-    - **Private Key**: the private key of the API key.
-    - **Description**: the description of the API key.
-    - **Action**: you can edit the description of the key or delete the key.
+### Manage linked clusters
 
-## Modify a Data App
+<!--TODO: add a new area: **Linked Cluster**-->
 
-You can rename a Data App, and manage its endpoints and API keys.
+You can add or remove linked clusters for a Data App. After you remove a linked cluster from, the endpoints in the Data App can still access other linked clusters.
```
Comment (qiancai) [inline tidb-cloud/data-service-manage-data-app.md:36]:
```suggestion
You can add or remove linked clusters for a Data App. After you remove a linked cluster, the endpoints in the Data App can still access other linked clusters.
```

File: tidb-cloud/data-service-manage-data-app.md [hunk 15]
```diff
@@ -17,35 +17,43 @@ To create a Data App for your project, perform the following steps:
...
 
-### Rename a Data App
+<!--TODO: add a new button: **Add Cluster**-->
 
-To rename a Data App, navigate to the [**Data Service**](https://tidbcloud.com/console/dataservice) page, and do one of the following:
+To add a linked cluster to a Data App, perform the following steps:
```
Comment (qiancai) [inline tidb-cloud/data-service-manage-data-app.md:40]:
```suggestion
To link a cluster to a Data App, perform the following steps:
```

File: tidb-cloud/data-service-manage-data-app.md [hunk 16]
```diff
@@ -17,35 +17,43 @@ To create a Data App for your project, perform the following steps:
...
 
-- In the left pane, click the name of your target Data App to view its details. In the **Basic Settings** area, modify the **Name** field and click **Save**.
+<!--TODO: add a new button: Linked Cluster > Action > Delete-->
+
+To remove a linked cluster from a Data App, perform the following steps:
+
+1. Navigate to the [**Data Service**](https://tidbcloud.com/console/dataservice) page of your project.
+2. In the left pane, locate your target Data App and click the name of your target Data App to view its details.
+3. In the **Linked Cluster** area, locate the target linked cluster you want to remove from the Data App, and click **Delete** in the **Action** column.
+4. In the displayed dialog box, click **I understand, remove** to confirm the removal.
```
Comment (qiancai) [inline tidb-cloud/data-service-manage-data-app.md:54]:
```suggestion
4. In the displayed dialog box, confirm the removal.
```

File: tidb-cloud/tidb-cloud-release-notes.md [hunk 17]
```diff
@@ -8,6 +8,16 @@ aliases: ['/tidbcloud/supported-tidb-versions','/tidbcloud/release-notes']
 
 This page lists the release notes of [TiDB Cloud](https://www.pingcap.com/tidb-cloud/) in 2023.
 
+## March 29, 2023
```
Comment (Oreoxmt) [inline tidb-cloud/tidb-cloud-release-notes.md:11]: @qiancai @heibaijian PTAL

File: tidb-cloud/tidb-cloud-roadmap.md [hunk 18]
```diff
@@ -34,8 +34,8 @@ The TiDB Cloud roadmap brings you what's coming in the near future, so you can s
     <td>In Chat2Query, you can either let AI generate SQL queries automatically or write SQL queries manually, and run SQL queries against databases without a terminal.</td>
   </tr>
   <tr>
-    <td>Support Data API.</td>
-    <td>Allow developers to read/write databases via data API.</td>
+    <td>✅ Support Data Service.</td>
```
Comment (Oreoxmt) [inline tidb-cloud/tidb-cloud-roadmap.md:37]: @qiancai @heibaijian PTAL

Comment (qiancai) [reply tidb-cloud/tidb-cloud-roadmap.md:37]: LGTM

File: tidb-cloud/tidb-cloud-release-notes.md [hunk 19]
```diff
@@ -8,6 +8,16 @@ aliases: ['/tidbcloud/supported-tidb-versions','/tidbcloud/release-notes']
...
+
+**General changes**
+
+- [Data Service (beta)](/tidb-cloud/data-service-overview.md) supports more fine-grained access control for Data Apps.
+
+    On the Data App details page, you can manage the linked clusters that your Data App can access and specify the role for each API key. The role controls whether the API key can read or write data to the linked clusters and can be set to ReadOnly or ReadAndWrite. This feature provides cluster-level and permission-level access control for Data Apps, giving you more flexibility for you to control the access scope according to your business needs.
```
Comment (qiancai) [inline tidb-cloud/tidb-cloud-release-notes.md:17]:
```suggestion
    On the Data App details page, now you can link clusters to your Data App and specify the role for each API key. The role controls whether the API key can read or write data to the linked clusters and can be set to `ReadOnly` or `ReadAndWrite`. This feature provides cluster-level and permission-level access control for Data Apps, giving you more flexibility to control the access scope according to your business needs.
```

File: tidb-cloud/tidb-cloud-release-notes.md [hunk 20]
```diff
@@ -8,6 +8,16 @@ aliases: ['/tidbcloud/supported-tidb-versions','/tidbcloud/release-notes']
...
+
+    For more details, see [Manage linked clusters](/tidb-cloud/data-service-manage-data-app.md#manage-linked-clusters) and [Manage API keys](/tidb-cloud/data-service-api-key.md).
```
Comment (qiancai) [inline tidb-cloud/tidb-cloud-release-notes.md:19]:
```suggestion
    For more information, see [Manage linked clusters](/tidb-cloud/data-service-manage-data-app.md#manage-linked-clusters) and [Manage API keys](/tidb-cloud/data-service-api-key.md).
```

File: tidb-cloud/tidb-cloud-release-notes.md (hunk 19 above)
Comment (qiancai) [reply tidb-cloud/tidb-cloud-release-notes.md:17]: reason: "you can manage the linked clusters" 看起来有歧义，可能理解为这个页面上用户可以运维集群

**Review by qiancai — APPROVED** [review]
other LGTM

**Pull request merged**